src/client/common/configSettings.ts

```diff
@@ -232,7 +239,8 @@ export class JupyterSettings implements IWatchableJupyterSettings {
     protected initialize(): void {
         const onDidChange = () => {
             const currentConfig = this._workspace.getConfiguration('jupyter', this._workspaceRoot);
-            this.update(currentConfig);
+            const pythonConfig = this._workspace.getConfiguration('python', this._workspaceRoot);
+            this.update(currentConfig, pythonConfig);
```

Comment:
Should we run this code everytime a configurataion item changes.
Shouldn't we do this only if Python poetryPath changes?

Reply:
It runs every time juptyer changes right now. Which is a mistake, it should run when poetryPath changes too.